exonum/src/messages/mod.rs

```diff
@@ -29,7 +29,63 @@
 //!                 |                                |
 //!                 V                                V
 //!              (drop)                           (drop)
+//!
+//! ```
+//!
+//! # Examples
+//!
+//! The procedure of creating a new signed message.
+//!
+//! ```
+//! use exonum::{
+//!     crypto::{self, Hash},
+//!     helpers::Height,
+//!     messages::{Status, Verified},
+//! };
+//!
+//! # fn send<T>(_: T) {}
+//! let keypair = crypto::gen_keypair();
+//! // Let we have a some Status message.
+//! let payload = Status {
+//!     height: Height(15),
+//!     last_hash: Hash::zero(),
+//! };
+//! // With a some keypair we can sign it and get trusted status message.
+//! let signed_payload = Verified::from_value(payload, keypair.0, &keypair.1);
+//! // Later we can convert it to raw signed message and sends by the network.
+//! let raw_signed_message = signed_payload.into_raw();
+//! send(raw_signed_message);
 //! ```
+//!
+//! The procedure of a signed message verification.
+//!
+//! ```
+//! use exonum::{
+//!     crypto::{self, Hash},
+//!     helpers::Height,
+//!     messages::{Status, Verified, SignedMessage, ExonumMessage},
+//! };
+//!
+//! # fn get_signed_message() -> SignedMessage {
+//! #   let keypair = crypto::gen_keypair();
+//! #   let payload = Status {
+//! #       height: Height(15),
+//! #       last_hash: Hash::zero(),
+//! #   };
+//! #   Verified::from_value(payload, keypair.0, &keypair.1).into_raw()
+//! # }
+//! // For example we have got a some signed message.
+//! let raw: SignedMessage = get_signed_message();
+//! // We know that this is one type of `ExonumMessage`, so we can try to
+//! // verify its signature and try to convert it into `ExonumMessage`.
```

Comment:
//! // You know that this is a type of `ExonumMessage`, so you can 
//! // verify its signature and convert it into `ExonumMessage`.